flatcar-linux/portage-stable
# bump(sys-libs/libseccomp): sync with upstream

Update libsseccomp from Gentoo:
https://gitweb.gentoo.org/repo/gentoo.git/tree/sys-libs/libseccomp/libseccomp-2.5.0.ebuild

Python check changed to python3_6 so it builds in Flatcar.

# How to use

To fully test libseccomp 2.5.0 with the Seccomp Notify feature, we need Linux 5.9-rc1. But basic non-regression tests can be done as follows.

# Testing done

After building a Flatcar image:
```
Flatcar Container Linux by Kinvolk developer (2605.0.0+2020-09-26-1247)
core@localhost ~ $ ldd /run/torcx/unpack/docker/bin/runc 
	linux-vdso.so.1 (0x00007fff6adeb000)
	libpthread.so.0 => /lib64/libpthread.so.0 (0x00007f877e7c9000)
	libseccomp.so.2 => /lib64/libseccomp.so.2 (0x00007f877e7a7000)
	libc.so.6 => /lib64/libc.so.6 (0x00007f877e5cf000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f877f1b2000)
core@localhost ~ $ ls -l /lib64/libseccomp.so* 
lrwxrwxrwx. 1 root root     19 Sep 26 13:01 /lib64/libseccomp.so -> libseccomp.so.2.5.0
lrwxrwxrwx. 1 root root     19 Sep 26 13:01 /lib64/libseccomp.so.2 -> libseccomp.so.2.5.0
-rwxr-xr-x. 1 root root 132968 Sep 26 12:35 /lib64/libseccomp.so.2.5.0
```

Then, I tested `/run/torcx/unpack/docker/bin/runc` both with and without a seccomp profile:
- Prepare a `rootfs` and `config.json` following instructions on https://github.com/opencontainers/runc#using-runc
- Run `/run/torcx/unpack/docker/bin/runc run c1`
- Check `cat /proc/self/status|grep -i seccomp` in the container: 2 when seccomp is enabled in config.json, or 0 when not enabled
- Test the following seccomp profile in `config.json`:
```
    "seccomp": {
      "defaultAction": "SCMP_ACT_ALLOW",
      "architectures": [
        "SCMP_ARCH_X86",
        "SCMP_ARCH_X32"
      ],
      "syscalls": [
        {
          "names": [
            "fchmod",
            "chmod"
          ],
          "action": "SCMP_ACT_ERRNO"
        }
      ]
    },
```
And check if it is enforced:
```
/ # touch /dev/shm/file
/ # chmod 777 /dev/shm/file
chmod: /dev/shm/file: Operation not permitted
```

---

Fixes https://github.com/flatcar-linux/Flatcar/issues/200
